### ISSUE TITLE
Do we need to stop gradient when computing `alpha_loss` and `policy_loss`?

### ISSUE THREAD

**Issue by fuyw** [young-geng/JaxCQL#3]
Hi Young,

I noticed that in the dopamine's implementation, it stops gradient for alpha when computing the `policy loss` to avoid the gradient changing the value of `alpha`.

```
alpha_value = jnp.exp(jax.lax.stop_gradient(log_alpha))
policy_loss = jnp.mean(alpha_value * action_log_prob - no_grad_q_value)
```
https://github.com/google/dopamine/blob/1cbed6c7c35163e73b0ee2f2d2bf032b057570dd/dopamine/jax/agents/sac/sac_agent.py#L170

Similarly, it stops gradient for the policy when computing the `alpha loss` to avoid affecting the policy.

```
alpha_loss = jnp.mean(log_alpha * jax.lax.stop_gradient(entropy_diff))
```
https://github.com/google/dopamine/blob/1cbed6c7c35163e73b0ee2f2d2bf032b057570dd/dopamine/jax/agents/sac/sac_agent.py#L175

**Comment by young-geng** [comment]
We don't need to do it here because we never add up the loss from different parts of our model. Instead, we keep a dictionary of all the individual losses, and take gradient for each one separately ([in line 281](https://github.com/young-geng/JaxCQL/blob/7ebaecf48e4f8a7e65fe9f0f03425385b0c742b6/JaxCQL/conservative_sac.py#L281)). This paradigm allows us to freely compose different networks without having to worry about extra gradient accumulated coming from other losses. The JIT compilation of Jax will eliminate all the repeated computations so the train step runs efficiently despite repeated calls to the loss function.

**Comment by fuyw** [comment]
Thank you. That's a nice design.
